lib/rules/indent.js

```diff
@@ -807,6 +891,95 @@ module.exports = {
             });
         }
 
+        /**
+        * Set the current unknown node if not set
+        * @param {ASTNode} node Node
+        * @returns {void}
+        */
+        function setCurrentUnknownNode(node) {
+            if (node.type in KNOWN_NODES) {
+                return;
+            }
+
+            if (currentUnknownNode === null) {
+                currentUnknownNode = node;
+            }
+        }
+
+        /**
+        * On exit of unknown node append node to like of outer most nodes
+        * @param {ASTNode} node Node
+        * @returns {void}
+        */
+        function saveOutermostUnknownNode(node) {
+            if (node.type in KNOWN_NODES) {
+                return;
+            }
+
+            if (currentUnknownNode === node) {
+                unknownNodes.push(node);
+                currentUnknownNode = null;
+            }
+        }
+
+        /**
+        * Ignore each line of an unknown node if its offset has no dependencies.
```

Comment:
I ignore the first token of the line within the unknown node. This way it's desired offset will be set to it's current offset.

If the offset is dependent on another token we have to find this token and make its offset depend on the first token of its line, since that token is already ignored.